src/components/ChatView.tsx

```diff
@@ -67,39 +70,60 @@ export default function ChatView (props: Props) {
       return msg.type !== MessageType.Caption
     })
 
+  const lastIndexOfMovedMessage = findLastIndex(
+    messages,
+    message => message.type === MessageType.MovedRoom
+  )
+  const currentRoomMessages = messages.slice(lastIndexOfMovedMessage)
+
+  const shownMessages = shouldShowOldMessages ? messages : currentRoomMessages
+
   return (
-    <div id="messages" onScroll={handleScroll}>
-      {messages.slice(-150).map((m, idx) => {
-        let hideTimestamp = false
-        const previousMessage = props.messages[idx - 1]
-        if (previousMessage) {
-          // TODO: Give all messages a userId for this to be meaningful
-          if (
-            (previousMessage as any).userId &&
-            (m as any).userId &&
-            (previousMessage as any).userId === (m as any).userId
-          ) {
-            const diff =
-              new Date(m.timestamp).getTime() -
-              new Date(previousMessage.timestamp).getTime()
-            // This is a bad way to calculate '3 minutes' and I should feel bad -em
-            if (diff < 1000 * 60 * 3) {
-              hideTimestamp = true
+    <>
+      <button
+        className="link-styled-button"
+        onClick={() => setShouldShowOldMessages(!shouldShowOldMessages)}
+      >
+        {shouldShowOldMessages ? 'Hide' : 'Show'} Older Messages
+      </button>
+      <div id="messages" onScroll={handleScroll}>
+        {shownMessages.slice(-150).map((m, idx) => {
+          let hideTimestamp = false
+          const previousMessage = props.messages[idx - 1]
+          if (previousMessage) {
+            // TODO: Give all messages a userId for this to be meaningful
+            if (
+              (previousMessage as any).userId &&
+          (m as any).userId &&
+          (previousMessage as any).userId === (m as any).userId
+            ) {
+              const diff =
+            new Date(m.timestamp).getTime() -
+            new Date(previousMessage.timestamp).getTime()
+              // This is a bad way to calculate '3 minutes' and I should feel bad -em
+              if (diff < 1000 * 60 * 3) {
+                hideTimestamp = true
+              }
             }
           }
-        }
 
-        const id = `message-${idx}`
-        return (
-          <MessageView
-            message={m}
-            key={id}
-            id={id}
-            hideTimestamp={hideTimestamp}
-            msgIndex={idx}
-          />
-        )
-      })}
-    </div>
+          const shouldShowInterstitial = m.type === MessageType.MovedRoom
+          const id = `message-${idx}`
+
+          return (
+            <>
+              {shouldShowInterstitial ? <hr key={id + '-interstitial'}/> : null}
```

Comment:
I opted to add this interstitial logic instead of changing the `MovedRoomView` here. The `MessageView` component doesn't really allow a message to interject an `hr` before the message. 

---

The other approach I considered was creating a dummy `MessageType` called something like `MessageType.Decorator` that took in some React content. Then we could interject decorator messages where we liked in the `messages` list for `hr`'s or other elements in the future. 

This also didn't necessarily feel right since these things weren't really messages, and it felt weird to extend the `MessageType` type to accomplish this approach.

---

Open to input on the "correct" approach for this code base.